# Description: fixes the chef-server daemons to disable Merb's json loading
# and sets the full path to the libraries for the merb apps for the api and
# webui. no plans for Upstream to change this.
Index: chef-0.9.8+dfsg.orig/chef-server-api/bin/chef-server
===================================================================
--- chef-0.9.8+dfsg.orig.orig/chef-server-api/bin/chef-server	2010-08-05 13:07:42.000000000 -0600
+++ chef-0.9.8+dfsg.orig/chef-server-api/bin/chef-server	2010-08-10 00:43:45.000000000 -0600
@@ -22,20 +22,20 @@
 
 require "rubygems"
 require "merb-core"
-
+Merb.disable(:json)
 # Load chef and chef-server-api from source rather than gem, if present
 $:.unshift(File.expand_path(File.dirname(__FILE__) + '/../../chef/lib/'))
 $:.unshift(File.expand_path(File.dirname(__FILE__) + '/../lib'))
 
 # Print the version if we have -v or --version
 if ARGV.any? { |arg| arg =~ /\-v|\-\-version/ }
-  require 'chef-server-api/version'
+  require '/usr/share/chef-server-api/lib/chef-server-api/version'
   puts "Chef Server (API) Version: #{ChefServerApi::VERSION}"
   exit 1
 end
 
 require 'chef'
-require 'chef-server-api'
+require '/usr/share/chef-server-api/lib/chef-server-api'
 
 # Ensure the chef gem we load is the same version as the chef server
 unless defined?(Chef)
Index: chef-0.9.8+dfsg.orig/chef-server-webui/bin/chef-server-webui
===================================================================
--- chef-0.9.8+dfsg.orig.orig/chef-server-webui/bin/chef-server-webui	2010-08-05 13:07:42.000000000 -0600
+++ chef-0.9.8+dfsg.orig/chef-server-webui/bin/chef-server-webui	2010-08-10 00:44:34.000000000 -0600
@@ -25,20 +25,20 @@
 
 require "rubygems"
 require "merb-core"
-
+Merb.disable(:json)
 # Load chef and chef-server-api from source rather than gem, if present
 $:.unshift(File.expand_path(File.dirname(__FILE__) + '/../../chef/lib'))
 $:.unshift(File.expand_path(File.dirname(__FILE__) + '/../lib'))
 
 # Print the version if we have -v or --version
 if ARGV.any? { |arg| arg =~ /\-v|\-\-version/ }
-  require 'chef-server-webui/version'
+  require '/usr/share/chef-server-webui/lib/chef-server-webui/version'
   puts "Chef Server (Web UI) Version: #{ChefServerWebui::VERSION}"
   exit 1
 end
 
 require 'chef'
-require 'chef-server-webui'
+require '/usr/share/chef-server-webui/lib/chef-server-webui'
 
 if ARGV[0] && ARGV[0] =~ /^[^-]/
   ARGV.push "-H"
